src/components/RepoSubmission.tsx

```diff
@@ -17,44 +21,37 @@ const RepoSubmission = () => {
   }
 
   const userName = String(user.user_metadata.user_name);
+
   const saveToDataBase = (repoUrl: string): void => {
     setIsSubmissionInProcess(true);
 
-    // todo: #5 import the submission function here instead
     setTimeout(() => {
       setIsSubmissionInProcess(false);
-      setSubmitted(true);
-      setButtonPlaceHolder("Close");
-      const { isValid, sanitizedUrl } = isValidRepoUrl(repoUrl.replace(/\s+/g, ""));
 
-      console.log("is valid: ", isValid);
-      if (isValid && userName) {
-        sendMessage(userName, sanitizedUrl);
-        ToastTrigger({ message: "Data Submitted", type: "success" });
-      }
+      const { isValid, sanitizedUrl } = isValidRepoUrl(repoUrl.replace(/\s+/g, ""));
 
       if (!isValid) {
         ToastTrigger({ message: "Invalid repo url", type: "error" });
+        return setIsSubmissionInProcess(false);
       }
 
-      if (!userName) {
+      if (!userName || userName === "undefined") {
```

Comment:
I think you can remove `!userName` because always is false.

Reply:
Ok, what about `userName === "undefined"`?

Reply:
Both of you are wrong for multiple reasons:
- because the user is coming from a hook, it can be null, it doesn't make sense to destructure properties from `user?.user_metadata.user_name` because `user_metadata` is a generic key value any type, that is why the linter is so aggressively repenting against that usage 
- second is that you `saveToDataBase` is an async function but it's not awaited, nor the callback, this needs to be a `useCallback` because it doesn't have the user context at all, so in that code block the variable doesn't actually exist returning a `never` type

I know the quality of the previous code is lacking, but if we start improving we should fix these errors rather than create a negative process of elimination on various types (SupabaseUser, never), making production error out if you click too often between screens :D 

IMHO do this last tho, as the review comments should make this fix simpler! 🍕

Reply:
Code quality  lacking is a fact but I don't really know how to optimize this code, maybe I should to revert back the way I found it :sweat_smile: